osu.Game/Screens/OnlinePlay/Components/OnlinePlayBackgroundScreen.cs

```diff
@@ -89,5 +86,47 @@ public override bool OnExiting(ScreenExitEvent e)
             this.MoveToX(0);
             return result;
         }
+
+        [LongRunningLoad]
+        private partial class OnlineBeatmapBackground : Background
+        {
+            private readonly int beatmapId;
+
+            public OnlineBeatmapBackground(int beatmapId)
+            {
+                this.beatmapId = beatmapId;
+            }
+
+            [BackgroundDependencyLoader]
+            private void load(BeatmapLookupCache lookupCache, LargeTextureStore textures, CancellationToken cancellationToken)
+            {
+                try
+                {
+                    APIBeatmap? beatmap = lookupCache.GetBeatmapAsync(beatmapId, cancellationToken).GetResultSafely();
```

Comment:
This is a vague and mostly unsubstantiated worry so no action required, but I do kinda worry if this is gonna hold up. I don't think we've done hard async-to-sync `.GetResultSafely()` calls in `[LongRunningLoad]` BDLs anywhere. Especially when API calls (hidden beneath task completion sources) are involved as well.

It seems that in practice this will be a cache hit anyway so probably not that relevant but I guess we'll see how well this does...

Reply:
When you say "hold up", in what way do you mean/from what angle? I ask because it's possible there's something I haven't considered.

We don't really _need_ to use `GetResultSafely()`. The primary thinking here was `OnlineBeatmapSetCover` is also a LRL and the "long" part is how `OnlineStore` behaves internally down this sync path. Given that this is also an online request of sorts, it sounded like it should behave similarly.

> in practice this will be a cache hit anyway so

I wouldn't rely on this though. Hard to say whether the behaviour will be identical in creating vs joining rooms, for example.